A87-grpc-metrics-xds-outlier-detection.md

```diff
@@ -0,0 +1,64 @@
+A87: gRPC OTel Metrics for xDS Outlier Detection
+---
+* Author(s): Pardhu Konakanchi, Sai Teja
+* Approver:
+* Status: Draft
+* Implemented in: Core, Go
+* Last updated: 03-04-25
+
+## Abstract
+
+This document proposes some new metrics that will be added in gRPC for xDS Client Outlier Detection
+
+## Background
+
+[A50: gRPC xDS Outlier Detection Support](https://github.com/grpc/proposal/blob/master/A50-xds-outlier-detection.md) is a spec for gRPC to support xDS Outlier Detection. The current implementation only offers debug and trace logging in terms of visibility, which can be insufficient for understanding and diagnosing decision making in large-scale production systems. 
+
+Using A79: Non-per-call Metrics Architecture, it is possible to add granular metrics to make visibility into outlier detection easy for service owners utilizing gRPC.
+
+### Related proposals: 
+* [A50: gRPC xDS Outlier Detection Support](https://github.com/grpc/proposal/blob/master/A50-xds-outlier-detection.md)
+* [A66: OpenTelemetry Metrics](https://github.com/grpc/proposal/blob/master/A66-otel-stats.md)
+* [A79: Non-per-call Metrics Architecture](https://github.com/grpc/proposal/blob/master/A79-non-per-call-metrics-architecture.md)
+
+## Proposal
+
+[A79’s](https://github.com/grpc/proposal/blob/master/A79-non-per-call-metrics-architecture.md) non-per-call metrics architecture fits perfectly into the metrics reporting solution. We propose to aim for parity where appropriate with [Envoy’s Outlier Detection metric collection](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#outlier-detection-statistics) to be collected in gRPC
+
+Outlier Detection metrics will have the following labels:
+
+| Name        | Disposition | Description |
+| ----------- | ----------- | ----------- |
+| grpc.target | required | Indicates the target we are running outlier detection on. (Same as the attribute defined in [A66](https://github.com/grpc/proposal/blob/master/A66-otel-stats.md).) |
+
+The following metrics will be exported:
+
+| Name          | Type  | Unit  | Labels  | Description |
+| ------------- | ----- | ----- | ------- | ----------- |
+|  grpc.xds_client.outlier_detection.ejections_enforced_total | Counter | {ejection} | 	grpc.target |	Total enforced ejections due to any outlier type |
```

Comment:
The prefix `grpc.xds_client.` is not appropriate here, since these metrics are going to be generated by the outlier detection policy, not by the XdsClient.

I think these should use the prefix `grpc.lb.` instead.